DashboardClient/DashboardClient.cpp

```diff
@@ -26,6 +26,213 @@ namespace Umati
 			std::shared_ptr<OpcUaTypeReader> pTypeReader)
 			: m_pDashboardDataClient(pDashboardDataClient), m_pPublisher(pPublisher), m_pTypeReader(pTypeReader)
 		{
+			this->startEventThread();
+		}
+
+		DashboardClient::~DashboardClient() {
+			this->stopEventThread();
+		}
+
+		void DashboardClient::startEventThread() {
+			if (m_eventThreadRunning)
+			{
+				LOG(INFO) << "EventThread Running";
+				return;
+			}
+
+			auto func = [this]() {
+				int cnt = 0;
+				while (this->m_eventThreadRunning) {
+					if(!m_eventqueue.empty()) {
+						IDashboardDataClient::StructureChangeEvent sce = m_eventqueue.front();
+						m_eventqueue.pop();
+						this->m_dynamicNodes.push_back(sce.refreshNode);
+						if(sce.nodeAdded || sce.referenceAdded) {
+							this->updateAddDataSet(sce.refreshNode);
+							this->Publish();
+						}
+						if(sce.nodeDeleted || sce.referenceDeleted) {
+							this -> updateDeleteDataSet(sce.refreshNode);
+							if(sce.nodeDeleted == true) {
+							auto search = browsedSimpleNodes.find(sce.refreshNode);
+								if(search != browsedSimpleNodes.end()) {
+									std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = search->second;
+									this->deleteAndUnsubscribeNode(*simpleNode);
+								}
+							}
+							this -> Publish();
+    					}
+					}
+					std::this_thread::sleep_for(std::chrono::milliseconds(100));
+				}
+			};
+			m_eventThreadRunning = true;
+			m_eventThread = std::thread(func);
+		}
+		void DashboardClient::stopEventThread() {
+			m_eventThreadRunning = false;
+			if (m_eventThread.joinable())
+			{
+				m_eventThread.join();
+			}
+		}
+		void DashboardClient::reloadDataSet(ModelOpcUa::NodeId_t nodeId) {
+			auto search = browsedSimpleNodes.find(nodeId);
+			if(search == browsedSimpleNodes.end()){
+				return;
+			}
+			std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = search->second;
+			auto childNodes = simpleNode->ChildNodes;
+			if(childNodes.empty()) { 
+				return;
+			}
+			auto child = childNodes.front();
+			std::shared_ptr<const ModelOpcUa::PlaceholderNode> placeholderNode = std::dynamic_pointer_cast<const ModelOpcUa::PlaceholderNode>(child);
```

Comment:
Why not make data model non-const?

Reply:
This involves changeing some classes in the model and i the codebase. I want to do this, but maybe it is better to do this in a separate task.